Tests/AsyncKitTests/Future+CollectionTests.swift

```diff
@@ -58,8 +60,54 @@ final class FutureCollectionTests: XCTestCase {
         try XCTAssertEqual(times2.wait(), [2, 3, 4, 7])
         XCTAssertThrowsError(try times2Badly.wait())
     }
+    
+    func testSequencedFlatMapEach() throws {
+        struct SillyRangeError: Error {}
+        var value = 0
+        let lock = Lock()
+        let collection = eventLoop.makeSucceededFuture([1, 2, 3, 4, 5, 6, 7, 8, 9])
+        let times2 = collection.sequencedFlatMapEach { int -> EventLoopFuture<Int> in
+            lock.withLock { value = Swift.max(value, int) }
+            guard int < 5 else { Thread.sleep(forTimeInterval: 2.0); return self.eventLoop.makeFailedFuture(SillyRangeError()) }
```

Comment:
Can we avoid this thread sleep?

Reply:
In this case, the use of `Thread.sleep()` is an explicit "yield to any thread", simulating the current `EventLoop` performing I/O; I'm not aware of a better way (or, honestly, _any_ other way) to guarantee that other `EventLoop`s have a (very) reasonable chance of getting scheduled if they have any events to service. That being said, @lukasa may know of a better way. (In fact, I kinda hope he does - I always learn something whenever Cory chimes in on this kind of thing! ❤️)

Reply:
The easiest way to try to guarantee that other event loops have had a chance to run is to enqueue a do-nothing work item onto all of them and wait for completion. That is:

```swift
var futures: [EventLoopFuture<Void>] = []
for loop in eventLoopGroup {
    futures.append(loop.submit({ () }))
}

return EventLoopFuture<Void>.whenAllComplete(futures, on: self.eventLoop).map { _ in
    int * 2
}
```

Reply:
@Lukasa But it's the negative "return failure and ensure no other event loops are running any of the other futures in the input array" case these tests care about; enqueuing work on all of them and waiting on the current one feels like it would confuse the question. However, if you say it doesn't do so, I gladly concede you understand this area far better than I 😅

Reply:
@Lukasa Implemented your solution, and it's considerably more elegant than I first realized to boot!